src/Tools/SemanticSearch/BuildTask/GenerateFilteredReferenceAssembliesTask.cs

```diff
@@ -149,9 +157,38 @@ internal void WriteApis(string outputFilePath, byte[] peImage)
         var newContent = $"# Generated, do not update manually{Environment.NewLine}" +
             string.Join(Environment.NewLine, apis);
 
+        if (RequireNoApiChanges)
+        {
+            var oldContent = "";
+
+            if (File.Exists(outputFilePath))
+            {
+                try
+                {
+                    oldContent = File.ReadAllText(outputFilePath, Encoding.UTF8);
+                }
+                catch (FileNotFoundException)
+                {
+                }
+                catch (Exception e)
+                {
+                    Log.LogWarning($"Unable to read '{outputFilePath}': {e.Message}");
```

Comment:
Should this just be an error + return? Otherwise we're just going to log a warning, then likely log a different error which might make folks not notice the original warning that caused it.